src/kernel/arch/x86_64/gdt/gdt.h

```diff
@@ -0,0 +1,23 @@
+// A purpose for GDT
+
+#pragma once
+
+// Entry of table
+
+struct gdt_entry
+{
+    u16int base_low;
```

Comment:
The types you are using in your struct are not defined at all.
Instead you should use the type that are defined in  `#include <stdint.h>`